app/assets/javascripts/index/query.js

```diff
@@ -312,10 +374,31 @@ OSM.Query = function (map) {
         .join(),
       geom = `geom(${bbox})`,
       radius = 10 * Math.pow(1.5, 19 - zoom),
-      here = `(around:${radius},${latlng})`,
-      enclosed = "(pivot.a);out tags bb",
-      nearby = `(node${here};way${here};);out tags ${geom};relation${here};out ${geom};`,
-      isin = `is_in(${latlng})->.a;way${enclosed};out ids ${geom};relation${enclosed};`;
+      here = `(around:${radius},${latlng})`;
+
+    let dateFilter = "";  // OverpassQL filter (dates >= 1000 CE)
+    let jsDateFilter = null;  // JavaScript filter (dates < 1000 CE)
+
+    if (map.timeslider) {
+      const currentDate = map.timeslider.getDate();
+      if (currentDate) {
+        if (isBeforeYear1000(currentDate)) {
+          // OverpassQL doesn't support dates < 1000 CE, filter in JavaScript instead
+          jsDateFilter = currentDate;
+        } else {
+          dateFilter = `(if: (!is_tag("start_date") || t["start_date"] <= "${currentDate}") && (!is_tag("end_date") || t["end_date"] > "${currentDate}"))`;
```

Comment:
For future reference, this code only works because `currentDate` is guaranteed to be a full YYYY-MM-DD date. If it ever becomes possible to set the date to something vaguer, then we’d need to compare to the next year:

```js
          let currentYear = currentDate.match(/^-?\d+/);
          currentYear &&= parseInt(currentYear[0], 10);
          dateFilter = `(if: (!is_tag("start_date") || t["start_date"] < "${currentYear + 1}") && (!is_tag("end_date") || t["end_date"] > "${currentDate}"))`;
```